tests/Functional/EventListener/TagListenerTest.php

```diff
@@ -38,15 +38,24 @@ public function testAnnotationTagsAreInvalidated()
     {
         $client = static::createClient();
 
-        $client->getContainer()->mock(
-            'fos_http_cache.cache_manager',
-            CacheManager::class
-        )
-            ->shouldReceive('supports')->andReturn(true)
-            ->shouldReceive('invalidateTags')->with(['all-items'])
-            ->shouldReceive('invalidateTags')->with(['item-123'])
-            ->shouldReceive('flush')->once()
+        $mock = $this->createMock(CacheManager::class);
+        $mock->expects($this->any())
+            ->method('supports')
+            ->willReturn(true)
         ;
+        $mock->expects($this->at(0))
+            ->method('invalidateTags')
+            ->with(['all-items'])
+        ;
+        $mock->expects($this->at(1))
+            ->method('invalidateTags')
+            ->with(['item-123'])
+        ;
+        $mock->expects($this->once())
+            ->method('flush')
```

Comment:
> Expectation failed for method name is equal to <string:flush> when invoked 1 time(s).

That's the reported failure. And I don't understand what that expectation is meant to do. https://github.com/FriendsOfSymfony/FOSHttpCacheBundle/blob/3dfbad1bdfc78d259a73359cc6131c4b7c31726a/src/EventListener/TagListener.php does not call `flush` on the CacheManager at all. So this cannot work.
Also this to me this `testTagsAreSetWhenCacheable` looks much more like a unit test and not a functional test. I don't see the point of using the container here and then setting a mock on it.
I thougt replacing services with mocks at runtime does not work with sf 4 anyway.

Reply:
Also I'm not sure if you can combine `$this->once()` and `$this->at(...)` in a single mock at all.

Reply:
> I thougt replacing services with mocks at runtime does not work with sf 4 anyway.

It was re-enabled in https://github.com/symfony/symfony/pull/24418 - as long as they are public and not initialized.

once and at look bad together, but i think we have it working elsewhere. will check with `at()`, maybe thats it.

the idea of the test is that it would ensure the tags actually end up in the response - only unit tests are not telling me that. when i locally dump the type of request, i get told that the listener sees a sub request and thats why it does not flush.